Documentation for the Authorization filter (#2992)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_assemblies/authorization/assembly-authorization-overview.adoc

:_mod-docs-content-type: ASSEMBLY

// file included in the following:
//
// kroxylicious-authorization/index.adoc

[id='assembly-authorization-overview-{context}']
= Authorization overview

[role="_abstract"]
The Authorization filter provides the ability for the proxy to enforce an authorization policy about Kafka resources.
These authorization checks are performed in addition to any authorization checks made by the broker itself.
For an action to be allowed, both the authorization filter and broker must decide that the action is allowed.

In general, the Authorization filter will make access decisions in the same manner as Kafka itself.
From the client's perspective, it will be impossible for it to distinguish between authorization enforced on the proxy and authorization enforced on the kafka cluster itself.

In order to use the authorization filter, the proxy must be configured in such a way that it knows the authenticated subject.

> If your applications are using TLS client authentication this may be as achieved using TODO: link to configuring subject builder

Reviewer: ```suggestion
If your applications use TLS client authentication, configure a TODO: link to subject builder
```
